src/esm_runscripts/prepare.py

```diff
@@ -672,6 +674,89 @@ def set_parent_info(config):
     return config
 
 
+def add_vcs_info(config):
+    """
+    Adds version control system information in a plain text yaml file under the
+    experiment log directory for both the models and the esm-tools.
+
+    Parameters
+    ----------
+    config : dict
+        The experiment configuration
+
+    Returns
+    -------
+    config : dict
+        The experiment configuration
+    """
+    exp_vcs_info_file = f"{config['general']['thisrun_log_dir']}/{config['general']['expid']}_vcs_info.yaml"
+    logger.debug("Experiment information is being stored for usage under:")
+    logger.debug(f">>> {exp_vcs_info_file}")
+    vcs_versions = {}
+    all_models = config.get("general", {}).get("models", [])
+    for model in all_models:
+        model_dir = config[model]["model_dir"]
+        if helpers.is_git_repo(model_dir):
+            vcs_versions[model] = helpers.get_all_git_info(model_dir)
+        else:
+            vcs_versions[model] = "Not a git-controlled model!"
```

Comment:
I hope no one is using a SVN based system. We support it but this will require extra refactoring 🥲

Reply:
Let's wait for someone to ask for SVN...none of the models I actively support uses anything aside from git.